clouddriver-google-common/src/main/groovy/com/netflix/spinnaker/clouddriver/googlecommon/deploy/GoogleApiException.java

```diff
@@ -19,40 +19,61 @@
 import com.google.api.client.googleapis.json.GoogleJsonError;
 import com.google.api.client.googleapis.json.GoogleJsonError.ErrorInfo;
 import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableList;
+import com.netflix.spinnaker.kork.annotations.NonnullByDefault;
 import java.util.Optional;
+import lombok.Value;
 
+@NonnullByDefault
 public class GoogleApiException extends RuntimeException {
   GoogleApiException(String message) {
     super(message);
   }
 
   static GoogleApiException fromGoogleJsonException(GoogleJsonResponseException e) {
-    Optional<ErrorInfo> optionalErrorInfo =
-        Optional.ofNullable(e.getDetails()).map(GoogleJsonError::getErrors)
-            .orElse(ImmutableList.of()).stream()
-            .findFirst();
-
-    if (optionalErrorInfo.isPresent()) {
-      ErrorInfo errorInfo = optionalErrorInfo.get();
-      String message =
-          String.format(
-              "Operation failed. Last attempt returned status code %s with error message %s and reason %s.",
-              e.getStatusCode(), errorInfo.getMessage(), errorInfo.getReason());
-      if ("resourceInUseByAnotherResource".equals(errorInfo.getReason())) {
-        return new ResourceInUseException(message);
+    ErrorDetails errorDetails = ErrorDetails.fromGoogleJsonException(e);
+    if (errorDetails.getReason().equals("resourceInUseByAnotherResource")) {
+      return new ResourceInUseException(errorDetails.toString());
+    }
+    return new GoogleApiException(errorDetails.toString());
+  }
+
+  @Value
+  private static class ErrorDetails {
+    private final int statusCode;
+    private final String message;
+    private final String reason;
+
+    static ErrorDetails fromGoogleJsonException(GoogleJsonResponseException e) {
+      Optional<ErrorInfo> optionalErrorInfo =
+          Optional.ofNullable(e.getDetails()).map(GoogleJsonError::getErrors)
+              .orElse(ImmutableList.of()).stream()
+              .findFirst();
```

Comment:
I'm not sure to be honest why it might have more than one error.  Maybe if it's a batch request with individual components, but I'm really just guessing at this point 🤷 

Given we've only been looking at the first one now I'm probably inclined to leave this until some use case arises where the other messages are useful.